### ISSUE TITLE
Why ref and definitions not working in this library

### ISSUE THREAD

**Issue by AbanoubAshraaf** [CareLuLu/react-native-web-jsonschema-form#202]
Hello , as in   [react-jsonschema-form](https://github.com/mozilla-services/react-jsonschema-form)  documentation if we write schema like this 
```
    const schema = {
  $schema: 'http://json-schema.org/draft-07/schema#',
  title: 'Schema fixed title',
  type: 'object',
  properties: {
    entrance: { $ref: '#/definitions/entrance' },
},
  definitions: {
    entrance: {
      $schema: 'http://json-schema.org/draft-07/schema',
      type: 'object',
      required: ['receptionExists'],
      properties: {
        entranceDoorTypes: {
          title: 'نوع أبواب المداخل', 
          type:'string'
 
        },
        receptionExists: {
          title: 'هل يوجد استقبال',
          type: 'boolean',
          default: false,
        },
        airConditioningExists: {
          title: 'هل يوجد تكييف',
          type: 'boolean',
          default: false,
        },
        lightingExists: {
          title: 'هل يوجد إضائة',
          type: 'boolean',
          default: false,
        },
        elevatorsExists: {
          title: 'هل يوجد مصاعد',
          type: 'boolean',
          default: false,
        },
        swimmingPoolExists: {
          title: 'هل يوجد مسبح ',
          type: 'boolean',
          default: false,
        },
        gardenExists: {
          title: 'هل يوجد حديقة ',
          type: 'boolean',
          default: false,
        },
        childrenPlaygrounds: {
          title: 'هل يوجد ملاعب أطفال',
          type: 'boolean',
          default: false,
        },
      },
    },
  },
}
```
it will produce ui like this :
![image](https://user-images.githubusercontent.com/47993303/104134516-c0ce2c80-5392-11eb-8669-0ade4c8118de.png)

this occur because while compiling when find any `ref `  key like this 
`    entrance: { $ref: '#/definitions/entrance' },
`
will go through  definition object inside schema and match the value of key  with keys of definition object inside schema 

the  first question here  :- Why this is not occurs with this library although it was inpired on `react-jsonschema-form` 
The second question here :- Why title  `Schema fixed title` not printed ?

**Comment by CareLuLu-Gabriel** [comment]
@AbanoubAshraaf this library currently doesn't support `definitions` and `references`. If you have the time, and can open a PR, that would be great!

In the meantime, you can write your schema without the definitions:
```js
const schema = {
  type: 'object',
  properties: {
    entrance: {
      type: 'object',
      properties: {
        entranceDoorTypes: {
          title: 'نوع أبواب المداخل', 
          type:'string'
        },
        // ...
        childrenPlaygrounds: {
          title: 'هل يوجد ملاعب أطفال',
          type: 'boolean',
          default: false,
        },
      },
    },
  },
};
```

If the only field of the schema is `entrace`, you can simplify the schema above to:
```js
const schema = {
  type: 'object',
  properties: {
    entranceDoorTypes: {
      title: 'نوع أبواب المداخل', 
      type:'string'
    },
    // ...
    childrenPlaygrounds: {
      title: 'هل يوجد ملاعب أطفال',
      type: 'boolean',
      default: false,
    },
  },
};
```

**Comment by AbanoubAshraaf** [comment]
Okay , Thank you for clarification  , let me do it , What about  `The second question here :- Why title Schema fixed title not printed ?`

**Comment by CareLuLu-Gabriel** [comment]
@AbanoubAshraaf  Sorry, I missed the second question.

**Re: Why title Schema fixed title not printed?**
We have no schema level title implemented.

I recommend printing the title separate from the form or sending a pull request. But if you really want to print the title with the form, I think this could work because any property can have a title.
```js
const schema = {
  type: 'object',
  properties: {
    objectWithSchemaTitle: {
      title: 'Schema fixed title',
      type: 'object',
      properties: {
        entrance: {
          title: 'Entrance',
          type: 'object',
          properties: {
            entranceDoorTypes: {
              title: 'نوع أبواب المداخل', 
              type:'string'
            },
            // ...
            childrenPlaygrounds: {
              title: 'هل يوجد ملاعب أطفال',
              type: 'boolean',
              default: false,
            },
          },
        },
      },
    },
  },
};
```

**Comment by AbanoubAshraaf** [comment]
I Think any from has type: object not print the title

**Comment by CareLuLu-Gabriel** [comment]
@AbanoubAshraaf you're right! I just published `v3.1.3` fixing the title issue with `object` type.
https://github.com/CareLuLu/react-native-web-jsonschema-form/commit/93ca6418e02f490a547c1cbe03e68913ae54b584
